docs/src/pldm.md

```diff
@@ -0,0 +1,268 @@
+The table below shows the inventory commands and firmware update commands are supported by PLDM stack as FD.
+
+| Command Name                   | Command Code | Direction | Requirement |
+|--------------------------------|--------------|-----------|-------------|
+| `QueryDeviceIdentifiers`       | `0x01`       | UA -> FD  | Mandatory   |
+| `GetFirmwareParameters`        | `0x02`       | UA -> FD  | Mandatory   |
+| `RequestUpdate`                | `0x10`       | UA -> FD  | Mandatory   |
+| `GetPackageData`               | `0x11`       | FD -> UA  | Mandatory   |
+| `GetDeviceMetaData`            | `0x12`       | UA -> FD  | Mandatory   |
+| `PassComponentTable`           | `0x13`       | UA -> FD  | Mandatory   |
+| `UpdateComponent`              | `0x14`       | UA -> FD  | Mandatory   |
+| `RequestFirmwareData`          | `0x15`       | FD -> UA  | Mandatory   |
+| `TransferComplete`             | `0x16`       | FD -> UA  | Mandatory   |
+| `VerifyComplete`               | `0x17`       | FD -> UA  | Mandatory   |
+| `ApplyComplete`                | `0x18`       | FD -> UA  | Mandatory   |
+| `GetMetaData`                  | `0x19`       | FD -> UA  | Mandatory   |
+| `ActivateFirmware`             | `0x1A`       | UA -> FD  | Mandatory   |
+| `GetStatus`                    | `0x1B`       | UA -> FD  | Mandatory   |
+| `CancelUpdateComponent`        | `0x1C`       | UA -> FD  | Mandatory   |
+| `CancelUpdate`                 | `0x1D`       | UA -> FD  | Mandatory   |
+
+The diagram below shows a complete PLDM firmware update sequence:
+```mermaid
+sequenceDiagram
+    participant UA as Update Agent
+    participant FD as Firmware Device
+    UA->>FD: QueryDeviceIdentifiers
+    FD-->>UA: DeviceIdentifiers Response
+    UA->>FD: GetFirmwareParameters
+    FD-->>UA: FirmwareParameters Response
+    UA->>FD: RequestUpdate
+    FD-->>UA: Update Response
+    UA->>FD: PassComponentTable
+    FD-->>UA: ComponentTable Response
+    UA->>FD: UpdateComponent
+    FD-->>UA: UpdateComponent Response
+    FD->>UA: RequestFirmwareData
+    UA-->>FD: FirmwareData Response
+    FD->>UA: TransferComplete
+    UA-->>FD: TransferComplete Response
+    Note over FD: Verifying component
+    FD->>UA: VerifyComplete
+    UA-->>FD: VerifyComplete Response
+    FD->>UA: ApplyComplete
+    UA-->>FD: ApplyComplete Response
+    UA->>FD: ActivateFirmware
+    FD-->>UA: ActivateFirmware Response
+    UA->>FD: GetStatus
+    FD-->>UA: Status Response
+```
+
+## Interface
+
+```Rust
+/// Trait representing a PLDM Firmware Update Service Manager.
+///
+/// This trait defines the necessary methods to start and stop the PLDM firmware update service.
+///
+/// # Methods
+///
+/// * `start_service` - Asynchronously starts the PLDM firmware update service.
+/// * `stop_service` - Asynchronously stops the PLDM firmware update service.
+///
+/// # Errors
+///
+/// Both methods return a `Result` which, on error, contains a `PldmServiceError`.
+pub trait PldmFwUpdateServiceMgr {
```

Comment:
It would be good here and earlier to specify which components are in the user application, user-mode APIs, kernel capsules, and kernel drivers.

That would help understand which traits are called from which contexts as well.